extensions/wikia/Helios/Client.class.php

```diff
@@ -51,9 +52,24 @@ public function request( $resourceName, $getParams = [], $postData = [], $extraR
 
 		$options = array_merge( $options, $extraRequestOptions );
 
+		/*
+		 * MediaWiki's MWHttpRequest class heavily relies on Messaging API
+		 * (wfMessage()) which happens to rely on the value of $wgLang.
+		 * $wgLang is set after $wgUser. On per-request authentication with
+		 * an access token we use MWHttpRequest before wgUser is created so
+		 * we need $wgLang to be present. With GlobalStateWrapper we can set
+		 * the global variable in the local, function's scope, so it is the
+		 * same as the already existing $wgContLang.
+		 */
+		global $wgContLang;
+		$wrapper = new GlobalStateWrapper( [ 'wgLang' => $wgContLang ] );
+
 		// Request execution.
 		/** @var \MWHttpRequest $request */
-		$request = \Http::request( $options['method'], $uri, $options );
+		$request = $wrapper->wrap( function() use ( $options, $uri ) {
+			return \Http::request( $options['method'], $uri, $options );
+		} );
```

Comment:
I think a better approach would be to inject an HTTP client object. This would mean moving this functionality to a wrapper class but result would be a much cleaner `\Wikia\Helios\Client::request`. 

It would also mean that you could decouple this class from MW which would make it eligible for inclusion in `/lib/Wikia` and make it easier to test. 

At some point, and this illustrates why, we'll need to move away from `Http:request` and use an [HTTP client](http://docs.guzzlephp.org/en/latest/) that we can inject.